### PR TITLE
Add implementationDetails to context

### PR DESCRIPTION
**Pull request by jonsnyder**
## Description

Send the alloy version as part of the xdm.  See https://github.com/adobe/xdm/pull/769

## Related Issue
https://jira.corp.adobe.com/browse/CORE-31964

## Motivation and Context

This will enable us to see which version customers are running.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All tests pass and I've made any necessary test changes.
- [x] I have run the Sandbox successfully.

### CODE REVIEW

**Comment by Aaronius** [comment]
Reminder to add this option to the extension and docs as well.

**Comment by Aaronius** [comment]
Actually, should `implementationDetails` be required?

**Comment by jfkhoury** [comment]
I think it should be required, but let’s bring it up with Justin and Rares

File: test/unit/specs/components/Context/implementationDetailsFactory.spec.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import versionFactory from "../../../../../src/components/Context/implementationDetailsFactory";
```
Comment (Aaronius) [inline test/unit/specs/components/Context/implementationDetailsFactory.spec.js]: I think you probably meant to name this `implementationDetailsFactory`.

**Pull request merged**